mdal/frmts/mdal_gdal.cpp

```diff
@@ -0,0 +1,589 @@
+/*
+ MDAL - Mesh Data Abstraction Library (MIT License)
+ Copyright (C) 2018 Peter Petrik (zilolv at gmail dot com)
+*/
+
+#include "mdal_gdal.hpp"
+#include <assert.h>
+#include <limits>
+#include <gdal.h>
+#include "ogr_api.h"
+#include "ogr_srs_api.h"
+#include "gdal_alg.h"
+
+#define MDAL_NODATA -9999
+static inline bool is_nodata( double val, double nodata = -9999.0, double eps = std::numeric_limits<double>::epsilon() ) {return fabs( val - nodata ) < eps;}
+
+void MDAL::GdalDataset::init( const std::string &dsName )
+{
+  mDatasetName = dsName;
+
+  // Open dataset
+  mHDataset = GDALOpen( dsName.data(), GA_ReadOnly );
+  if ( !mHDataset ) throw MDAL_Status::Err_UnknownFormat;
+
+  // Now parse it
+  parseParameters();
+  parseProj();
+}
+
+void MDAL::GdalDataset::parseParameters()
+{
+  mNBands = GDALGetRasterCount( mHDataset );
+  if ( mNBands == 0 ) throw MDAL_Status::Err_InvalidData;
+
+  GDALGetGeoTransform( mHDataset, mGT ); // in case of error it returns Identid
+
+  mXSize = GDALGetRasterXSize( mHDataset ); //raster width in pixels
+  if ( mXSize == 0 ) throw MDAL_Status::Err_InvalidData;
+
+  mYSize = GDALGetRasterYSize( mHDataset ); //raster height in pixels
+  if ( mYSize == 0 ) throw MDAL_Status::Err_InvalidData;
+
+  mNPoints = mXSize * mYSize;
+  mNVolumes = ( mXSize - 1 ) * ( mYSize - 1 );
+}
+
+void MDAL::GdalDataset::parseProj()
+{
+  char *proj = const_cast<char *>( GDALGetProjectionRef( mHDataset ) );
+  if ( proj != nullptr )
+  {
+    mProj = std::string( proj );
+  }
+}
+
+/******************************************************************************************************/
+
+bool MDAL::LoaderGdal::meshes_equals( const MDAL::GdalDataset *ds1, const MDAL::GdalDataset *ds2 ) const
+{
+  return ( ( ds1->mXSize == ds2->mXSize ) &&
+           ( ds1->mYSize == ds2->mYSize ) &&
+           ( is_nodata( ds1->mGT[0], ds2->mGT[0] ) ) &&
```

Comment:
can we rename it to something like fuzzy_compare? is_nodata() is very non-intuitive